R/create_codemeta.R

```diff
@@ -14,155 +14,198 @@
 #' cm$keywords <- list("metadata", "ropensci")
 #' }
 #' @importFrom jsonlite read_json
-create_codemeta <- function(pkg = ".",
-                            root = ".",
-                            id = NULL,
-                            use_filesize = TRUE,
-                            force_update =
-                              getOption("codemeta_force_update", TRUE),
-                            verbose = TRUE,
-                            ...) {
+create_codemeta <- function(
+  pkg = ".",
+  root = ".",
+  id = NULL,
+  use_filesize = TRUE,
+  force_update =
+    getOption("codemeta_force_update", TRUE),
+  verbose = TRUE,
+  ...
+) {
+
   ## looks like we got a package name/path or Description file
   if (is.character(pkg)) {
+
     root <- get_root_path(pkg)
 
+    # Set string constants
+    json_file <- get_file("codemeta.json", root)
 
     ## no cm provided, but codemeta.json found in pkg
-    if (file.exists(get_file("codemeta.json", root))) {
+    cm <- if (file.exists(json_file)) {
+
```

Comment:
I'd prefer the assignment to be inside the in and else, not outside of them. :-)

Reply:
In general or only in this case? In general, I prefer to do the assignment outside, because to me `variable <-` is already a duplication when used in both, the if and the else branch. There is a risk that you use different variable names, such as in
```
if (condition) {
  my_long_variable_name <- "a"
} else {
  my_long_Variable_name <- "b"
}
```
To me, the following looks more clean and is less error-prone:
```
my_long_variable_name <- if (condition) {
  "a"
} else {
  "b"
}
```
However, a prerequisite is that the bodies of the if and else branches are small, in the best case one line only...
So, do you want me to undo this change? I have already used the "outside assignment" at many other places in the code...

Reply:
Ok then I'm convinced!

Reply:
So keep doing that ;-)

Reply:
Another advantage: when doing the assignment outside, the value of `NULL` is automatically assigned to the variable if there is no else branch, just as stated in help("if"):
> if  returns the value of the expression evaluated, or NULL invisibly if none was (which may happen if there is no else).

So, instead of 
```
if (condition) {
  result <- "a"
} else {
  result <- NULL
}
```
(a construct that is used a lot in the existing code), you can simply write
```
result <- if (condition) {
  "a"
} # otherwise NULL implicitly
```
Well, this might not be obvious to everyone and maybe we should prefer
```
result <- if (condition) {
  "a"
} else {# otherwise NULL implicitly
```

Reply:
😮 👏